src/devices/slacky_diy.ts

```diff
@@ -0,0 +1,791 @@
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentTier2SummDelivered !== undefined) {
+                        const data = msg.data.currentTier2SummDelivered;
+                        result.tariff2 = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+            {
+                cluster: "seMetering",
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentTier3SummDelivered !== undefined) {
+                        const data = msg.data.currentTier3SummDelivered;
+                        result.tariff3 = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+            {
+                cluster: "seMetering",
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentTier4SummDelivered !== undefined) {
+                        const data = msg.data.currentTier4SummDelivered;
+                        result.tariff4 = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+            {
+                cluster: "seMetering",
+                type: ["attributeReport", "readResponse"],
+                convert: (model, msg, publish, options, meta) => {
+                    const result: KeyValueAny = {};
+                    if (msg.data.currentSummDelivered !== undefined) {
+                        const data = msg.data.currentSummDelivered;
+                        result.tariff_summ = (Number.parseInt(data) / energyDivisor) * energyMultiplier;
+                    }
+                    return result;
+                },
+            },
+        ];
+        return {
+            exposes,
+            fromZigbee,
+            toZigbee,
+            isModernExtend: true,
+        };
+    },
+    powerSupplyParam: (): ModernExtend => {
+        const exposes: Expose[] = [
+            e.numeric("voltage", ea.STATE_GET).withUnit("V").withDescription("Voltage"),
+            e.numeric("current", ea.STATE_GET).withUnit("A").withDescription("Current"),
+            e.numeric("power", ea.STATE_GET).withUnit("kW").withDescription("Power"),
+        ];
+        const toZigbee: Tz.Converter[] = [
+            {
+                key: ["voltage"],
+                convertGet: async (entity, key, meta) => {
+                    await entity.read("haElectricalMeasurement", ["rmsVoltage", "acVoltageMultiplier", "acVoltageDivisor"]);
```

Comment:
Why also read the divisor and multiplier here? I think you can re-use https://github.com/Koenkk/zigbee-herdsman-converters/blob/8d21b389364565c8766538b03c02cb0397a5942c/src/converters/toZigbee.ts#L1914